crawler/helpers.py

```diff
@@ -144,52 +180,199 @@ def parse_csv(centre: Dict[str, str]) -> Tuple[str, List[str], List[Dict[str, st
 
-    return files_with_time[sorted(files_with_time, reverse=True)[0]]
+    # return the latest one
+    latest_file_name = files_with_time[sorted(files_with_time, reverse=True)[0]]
+    logger.info(f"Latest file: {latest_file_name}")
+
+    return latest_file_name
+
+
+def merge_daily_files(centre: Dict[str, str]) -> str:
+    """Merge all the daily incremental files of the centre into one 'master' file. The master
+    file's name is created by appending '_master' to the latest CSV file name.
+
+    Arguments:
+        centre {Dict[str, str]} -- the centre in question
+
+    Raises:
+        CentreFileError: raised when no field names are found in the CSV file
+
+    Returns:
+        str -- the name of the master file created
+    """
+    logger.info(f"Merging daily files of {centre['name']}")
+
+    # get list of files
+    centre_files = get_files_in_download_dir(centre, "sftp_file_regex")
+    logger.debug(f"{len(centre_files)} to merge into master file")
+
+    # get the latest file name to use for the master name
+    latest_file_name = get_latest_csv(centre, "sftp_file_regex")
+
+    master_file_name = f"{latest_file_name[:-4]}_master.csv"
+    with open(f"{get_download_dir(centre)}{master_file_name}", "w") as master_csv:
+        field_names_written = False
+        for filename in centre_files:
+            logger.debug(f"Merging {filename} into {master_file_name}")
+            with open(f"{get_download_dir(centre)}{filename}", "r", newline="") as daily_file:
+                csvreader = DictReader(daily_file)
+
+                if csvreader.fieldnames is None:
+                    raise CentreFileError("Field names required in CSV file")
+
+                #  write header
+                if not field_names_written:
+                    writer = DictWriter(master_csv, fieldnames=csvreader.fieldnames)
+                    writer.writeheader()
+                    field_names_written = True
+
+                # copy data
+                for row in csvreader:
+                    writer.writerow(row)
+
+    logger.info(f"{master_file_name} created")
+
+    return master_file_name
+
+
+def upload_file_to_sftp(config: ModuleType, centre: Dict[str, str], filename: str) -> None:
+    """Uploads the given file to the centre's SFTP write folder.
+
+    Arguments:
+        config {ModuleType} -- app config
+        centre {Dict[str, str]} -- the centre in question
+        filename {str} -- filename of file to be uploaded
+    """
+    logger.info(f"Uploading {filename} to {centre['sftp_root_write']}")
+
+    # disable host key checking:
+    #   https://bitbucket.org/dundeemt/pysftp/src/master/docs/cookbook.rst#rst-header-id5
+    cnopts = pysftp.CnOpts()
+    cnopts.hostkeys = None
+
+    sftp_host = config.SFTP_HOST  # type: ignore
+    sftp_port = config.SFTP_PORT  # type: ignore
+    sftp_user = config.SFTP_WRITE_USER  # type: ignore
+    sftp_password = config.SFTP_WRITE_PASSWORD  # type: ignore
+
+    with pysftp.Connection(
```

Comment:
This sftp connection code looks shared with that in download_csv_files. Can it be extracted easily?